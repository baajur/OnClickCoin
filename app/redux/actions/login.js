import { LOGIN } from './actionTypes';

// /*
//  * action creators
//  */

export default function login(username) {
  console.log('login action called');
  console.log('username is ', username);
  return { type: LOGIN, payload: { username } };
}
