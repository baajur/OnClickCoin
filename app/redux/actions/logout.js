import { LOGOUT } from './actionTypes';

export default function logout() {
  console.log('logout action called');
  return { type: LOGOUT };
}
