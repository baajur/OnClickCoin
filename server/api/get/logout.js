import { Router } from 'express';

export default () => {
  const app = Router();

  app.get('api/logout', (request, response) => {
    request.logout();
    response.send('/');
  });

  return app;
};
