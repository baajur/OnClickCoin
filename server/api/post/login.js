import { Router } from 'express';
const passport = require('passport');

export default () => {
  const app = Router();

  app.post(
    '/api/login',
    passport.authenticate('local', {
      successRedirect: 'login',
      failureFlash: true,
    }),
  );

  return app;
};
