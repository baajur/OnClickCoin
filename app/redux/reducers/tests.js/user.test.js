import userReducer from '../user';
import login from '../../actions/login';
import logout from '../../actions/logout';

describe('userReducer', () => {
  it('returns the initial state', () => {
    expect(userReducer(undefined, {})).toMatchSnapshot();
  });

  it('handles the login action', () => {
    expect(userReducer({}, login('admin'))).toMatchSnapshot();
  });

  it('handles the logout action', () => {
    expect(userReducer({}, logout())).toMatchSnapshot();
  });
});
