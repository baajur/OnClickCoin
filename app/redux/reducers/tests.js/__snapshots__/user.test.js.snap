// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userReducer handles the login action 1`] = `
Object {
  "username": "admin",
}
`;

exports[`userReducer handles the logout action 1`] = `
Object {
  "username": Object {
    "username": "anonymous",
  },
}
`;

exports[`userReducer returns the initial state 1`] = `
Object {
  "username": "anonymous",
}
`;
